packages/twenty-front/src/hooks/useSortedArray.ts

```diff
@@ -0,0 +1,39 @@
+import { useRecoilValue } from 'recoil';
+
+import { tableSortFamilyState } from '@/activities/states/tableSortFamilyState';
+import { OrderBy } from '@/object-metadata/types/OrderBy';
+
+export const useSortedArray = <T>(
+  arrayToSort: T[],
+  { tableId, initialFieldName }: { tableId: string; initialFieldName: string },
+): T[] => {
+  const sortConfig = useRecoilValue(
+    tableSortFamilyState({ tableId, initialFieldName }),
+  );
+
+  const sortTableData = (
+    data: T[],
+    columnKey: keyof T,
+    order: OrderBy | null,
+  ): T[] => {
+    return data.sort((a: T, b: T) => {
+      if (typeof a[columnKey] === 'string') {
```

Comment:
We shouldn't try to guess the type from JS since it can be unreliable. Instead we should give this hook metadata about our table so that it knows how to behave for each field